pkg/reconciler/space/resources/namespace.go

```diff
@@ -48,5 +48,8 @@ func MakeNamespace(space *v1alpha1.Space) (*v1.Namespace, error) {
 					managedByLabel:      "kf",
 				}),
 		},
+		Spec: v1.NamespaceSpec{
+			Finalizers: []v1.FinalizerName{"kubernetes"},
```

Comment:
I don't think we should be adding this, instead we should ignore the result. IIRC, if a finalizer is added it's the responsibility of the adder to remove it or else the resource won't be garbage collected. Adding it here might prevent k8s from doing that or whatever tasks need to be run by k8s when a namespace is created.

Reply:
SGTM